cpp/src/Slice/Parser.cpp

```diff
@@ -952,31 +953,40 @@ Slice::Contained::isDeprecated(bool checkParent) const
 {
     const string prefix1 = "deprecate";
     const string prefix2 = "deprecated";
-    string metadata;
     ContainedPtr parent = checkParent ? dynamic_pointer_cast<Contained>(_container) : nullptr;
 
-    return (findMetadata(prefix1, metadata) || (parent && parent->findMetadata(prefix1, metadata))) ||
-           (findMetadata(prefix2, metadata) || (parent && parent->findMetadata(prefix2, metadata)));
+    return (hasMetadata(prefix1) || (parent && parent->hasMetadata(prefix1))) ||
```

Comment:
With the updated `hasMetadata` using "deprecate" matches both, but I guess is not intentional.

Reply:
It would be nice to just support one flavor for this metadata.

Reply:
Yes, the loosening of `findsMetadata` was only going to be temporary, so I didn't want to cause more trouble for future me!

I think that "only supporting one" is the future goal. As of 3.7 we only supported `deprecate`, but with `3.8` we wanted to switch it to `deprecated`. That's a pretty negligible change though, so we didn't want to annoy users. In 3.9 I image we'll finalize the switch, or at least deprecate `deprecate`.